File: code/section_iii/3_2_passing_multiple_parameters.py
from sqlalchemy import create_engine, text

engine = create_engine('sqlite:///thunderbird_manufacturing.db')
conn = engine.connect()

stmt = text("SELECT * FROM CUSTOMER WHERE STATE = :state AND CATEGORY = :category")
results = conn.execute(stmt, state='TX', category='COMMERCIAL')

for r in results:
    print(r)

File: code/section_iii/3_2_passing_multiple_parameters.R
library(DBI)
library(RSQLite)

db <- dbConnect(SQLite(), dbname='thunderbird_manufacturing.db')

sql <- "SELECT * FROM CUSTOMER WHERE STATE = ? AND CATEGORY = ?"

my_query <- dbSendQuery(db, sql, params = list('TX', 'COMMERCIAL'))
my_data <- dbFetch(my_query, n = -1)

dbClearResult(my_query)
remove(my_query)
dbDisconnect(db)
print(my_data)